sys/random/fortuna/fortuna.h

```diff
@@ -0,0 +1,226 @@
+*         found at https://crypto.stackexchange.com/a/5736.
+ */
+#ifndef FORTUNA_AES_KEY_SIZE
+#define FORTUNA_AES_KEY_SIZE        (16U)
+#endif
+
+/**
+ * @brief Use a mutex to lock concurrent access when running sensitive
+ *        operations in multi-threaded applications.
+ */
+#ifndef FORTUNA_LOCK
+#define FORTUNA_LOCK                (1)
+#endif
+
+/**
+ * @brief When set, add additional instructions to clear temporary variables
+ *        to prevent leaking sensitive information.
+ */
+#ifndef FORTUNA_CLEANUP
+#define FORTUNA_CLEANUP             (1)
+#endif
+
+/**
+ * @brief Generator counter and key.
+ */
+typedef struct {
+    uint8_t key[32];
+    union {
+        struct {
+            uint64_t l;
+            uint64_t h;
+        } split;
+        uint8_t bytes[16];
+    } counter;
+} fortuna_generator_t;
+
+/**
+ * @brief Pool for storing entropy.
+ */
+typedef struct {
+    sha256_context_t ctx;
+    uint32_t len;
+} fortuna_pool_t;
+
+/**
+ * @brief The Fortuna state.
+ */
+typedef struct {
+    fortuna_generator_t gen;
+    fortuna_pool_t pools[FORTUNA_POOLS];
+    uint32_t reseeds;
+#if FORTUNA_RESEED_INTERVAL > 0
+    uint64_t last_reseed;
+#endif
+#if FORTUNA_LOCK
+    mutex_t lock;
+#endif
+} fortuna_state_t;
+
+/**
+ * @brief Type definition of a Fortuna seed file.
+ */
+typedef uint32_t fortuna_seed_t[FORTUNA_SEED_SIZE];
+
+/**
+ * @brief   Initialize the Fortuna PRNG state.
+ *
+ * It is possible to use this method to clear an existing state.
+ *
+ * @param[in] state         PRNG state
+ * @return                  zero on succesful initialization
+ * @return                  non-zero on error
+ */
+int fortuna_init(fortuna_state_t *state);
+
+/**
+ * @brief   Read random bytes from the PRNG. The number of bytes may not exceed
+ *          FORTUNA_RESEED_LIMIT bytes.
+ *
+ * @param[in] state         PRNG state
+ * @param[out] out          pointer to buffer
+ * @param[in] bytes         number of bytes to write in buffer
+ * @return                  zero on success
+ * @return                  -1 on reading more that FORTUNA_RESEED_LIMIT bytes
+ * @return                  -2 on reading from unseeded PRNG
+ * @returned                -3 on other error
+ */
+int fortuna_random_data(fortuna_state_t *state, uint8_t *out, size_t bytes);
+
+/**
+ * @brief   Add a entropy of a random event to one PRNG pool. The pool must
+ *          exist and the source length must be 1-32 bytes.
+ *
+ * @param[in] state         PRNG state
+ * @param[in] data          pointer to entropy source
+ * @param[in] length        length of entropy source
+ * @param[in] source        source identifier (each source has its own ID)
+ * @param[in] pool          pool number to add entropy to
+ * @return                  zero on success
+ * @return                  -1 on zero bytes or more than 32 bytes
+ * @return                  -2 on invalid pool number
+ */
+int fortuna_add_random_event(fortuna_state_t *state, uint8_t *data,
```

Comment:
same here (and below) consider making input params (pointers) const if possible.

Reply:
Same.